File: testdata/bench/filter_map.lua
local records = {}
local n = 100000

for i = 0, n - 1 do
    table.insert(records, {
        id = i,
        value = (i * 19) % 500,
        active = i % 3 == 0,
    })
end

local result = {}

for _, r in ipairs(records) do
    if r.active then
        if r.value > 100 then
            table.insert(result, {
                id = r.id,
                score = r.value * 2,
            })
        end
    end
end

local out = #result
print(out)

File: testdata/bench/filter_map.js
var records = [];
var n = 100000;

for (var i = 0; i < n; i++) {
    records.push({
        id: i,
        value: (i * 19) % 500,
        active: i % 3 === 0,
    });
}

var result = [];

for (var i = 0; i < records.length; i++) {
    var r = records[i];
    if (!r.active) {
        continue;
    }
    if (r.value > 100) {
        result.push({
            id: r.id,
            score: r.value * 2,
        });
    }
}

var out = result.length;
console.log(out);
